lib/v2/screens/explore_screens/vote_screens/proposals/components/vote_amount_label/interactor/usecases/get_vote_use_case.dart

```diff
@@ -1,8 +1,21 @@
 import 'package:async/async.dart';
+import 'package:hive/hive.dart';
 import 'package:seeds/v2/datasource/remote/api/proposals_repository.dart';
+import 'package:seeds/v2/datasource/remote/model/vote_model.dart';
 
 class GetVoteUseCase {
-  Future<Result> run(int proposalId, String account) {
-    return ProposalsRepository().getVote(proposalId, account);
+  Future<Result> run(int proposalId, String account) async {
+    var box = await Hive.openBox<VoteModel>("votes.1.box");
```

Comment:
i dont like that we talk to Hive directly. 
This should be hidden behind a repository 

We are creating strong dependencies to hive when we do this.

Reply:
I'm not sure if these Hive boxes should already be instantiated in the class, maybe we need a `init(bo1, box2 ..)` function in the main file, and there pass all the boxes to the class not sure if this also works, although it is also not very clear to me if there will be scenarios where these boxes should be closed.

Reply:
From Reso coder.
> To keep the code clean, it's probably a wise idea to open the Box from only a single place and then to get it using Hive.box('name'). Also, to prevent holding unnecessary data in memory, you can close the Box when you're not going to need it anymore. Hive also has a handy method to close all boxes. It's a good practice to do this before the app exits, although as per the official documentation, it's not really necessary to do so.
> 
> Before your application exits, you should call Hive.close() to close all open boxes. Don’t worry if the app is killed before you close Hive, it doesn’t matter.
> 

Maybe not a init function maybe we can put the `Hive.box('name')` inside the repo and then just pass the name of the box in constructor. Exmaple:

`CacheRepository<VoteModel>("votes.1.box")`

Reply:
@gguijarro-c-chwy any thoughts? or I just leave this solution for now?

Reply:
If this works, we can leave this solution for now. I havent had time to look deeper into this. 
Later on ill take a closer took.